nixos/modules/services/web-apps/reposilite.nix

```diff
@@ -0,0 +1,253 @@
+              Embedded databases such as SQLite or H2 don't support truly concurrent connections, so the value will be always 1 for them if selected.
+            '';
+            default = 1;
+            example = 3;
+            type = types.int;
+          };
+
+          compressionStrategy = mkOption {
+            description = ''
+              Select compression strategy used by this instance.
+              Using 'none' reduces usage of CPU & memory, but ends up with higher transfer usage.
+              GZIP is better option if you're not limiting resources that much to increase overall request times.
+              Available strategies: none, gzip
+            '';
+            default = "none";
+            example = "gzip";
+            type = types.enum [
+              "none"
+              "gzip"
+            ];
+          };
+
+          idleTimeout = mkOption {
+            description = "Default idle timeout used by Jetty";
+            default = 30000;
+            example = 120000;
+            type = types.int;
+          };
+
+          bypassExternalCache = mkOption {
+            description = ''
+              Adds cache bypass headers to each request from /api/* scope served by this instance.
+              Helps to avoid various random issues caused by proxy provides (e.g. Cloudflare) and browsers.
+            '';
+            default = true;
+            type = types.bool;
+          };
+
+          cachedLogSize = mkOption {
+            description = "Amount of messages stored in cached logger.";
+            default = 32;
+            example = 64;
+            type = types.int;
+          };
+
+          defaultFrontend = mkOption {
+            description = "Enable default frontend with dashboard.";
+            default = true;
+            type = types.bool;
+          };
+
+          basePath = mkOption {
+            description = ''
+              Set custom base path for Reposilite instance.
+              It's not recommended to mount Reposilite under custom base path and you should always prioritize subdomain over this option.
+            '';
+            default = "/";
+            example = "/maven";
+            type = types.str;
+          };
+
+          debugEnabled = mkOption {
+            description = "Debug mode";
+            default = false;
+            type = types.bool;
+          };
+
+        };
+      };
+      description = "Configuration written to the config file for Reposilite.";
+      default = { };
+    };
+
+    tokens = mkOption {
+      description = ''
+        Tokens to use for authentication.
+        Passed to Reposilite via the `--token` option, in the format <name>:<value>.
+        It is recommended to add a least one token here, so that you can access the dashboard (more can be added from there).
+      '';
+      example = ''
+        [
+          "admin:password"
+        ]
+      '';
+      type = types.listOf types.str;
+    };
```

Comment:
It would be great if we could somehow make this take a file or collection of files instead of text. Text will end up the nix store, available for all users on the system.